### PR TITLE
Feature app

### PR DESCRIPTION
**Pull request by somoye123**
Implement data structures necessary to support link submissions and commenting on Reddit

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+
+  validates :title, presence: true
+  validates :url, presence: true
+  validates :user_id, presence: true
```
Comment (maelfosso) [inline app/models/post.rb:7]: Kindly change `:user_id` by `:user` to make it works.

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  validates :body, presence: true
+  validates :user_id, presence: true
+  validates :post_id, presence: true
```
Comment (maelfosso) [inline app/models/comment.rb:7]: Kindly directly use `:user` and `:post` to make it works

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,

Thanks for the effort put in this project.
However, there are some issues.
Please, read the comment below. I suggest you use `rails console` to test the validity of object create from your model.

Once done, submit a new code review request.
Thanks

Mael FOSSO

**Comment by uryelah** [comment]
Hi **Somoye Somoye & Uduak Essien**, I'm Sarah your reviewer for this PR 👩‍🚀 

Good job with the fixes guys!

On an [Optional] note, why don't you have a bit more fun with the validations? 
- [ ] Try to validate if the email string is actually in the email format, you can use both regular expressions or rails inbuilt classes for that
- [ ] Validate the length of the body and title
- [ ] Validate the password so it's strong enough, can you make sure it's at least 6 characters, has letters and numbers and at least one symbol?

-------

## **Your project is approved! 🎊 🥅 💛**

**You may close this Pull Request and merge the changes**


---------

    
    
**Keep being awesome devs and good luck with your next project!**

---------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Review by uryelah — APPROVED** [review]
Hi **Somoye Somoye & Uduak Essien**, I'm Sarah your reviewer for this PR 👩‍🚀 

Good job with the fixes guys!

On an [Optional] note, why don't you have a bit more fun with the validations? 
- [x] Try to validate if the email string is actually in the email format, you can use both regular expressions or rails inbuilt classes for that
- [x] Validate the length of the body and title
- [x] Validate the password so it's strong enough, can you make sure it's at least 6 characters, has letters and numbers and at least one symbol?

-------

## **Your project is approved! 🎊 🥅 💛**

**You may close this Pull Request and merge the changes**


---------

    
    
**Keep being awesome devs and good luck with your next project!**

---------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Comment by acushlakoncept** [comment]
> Hi **Somoye Somoye & Uduak Essien**, I'm Sarah your reviewer for this PR 👩‍🚀
> 
> Good job with the fixes guys!
> 
> On an [Optional] note, why don't you have a bit more fun with the validations?
> 
> * [x]  Try to validate if the email string is actually in the email format, you can use both regular expressions or rails inbuilt classes for that
> * [x]  Validate the length of the body and title
> * [x]  Validate the password so it's strong enough, can you make sure it's at least 6 characters, has letters and numbers and at least one symbol?
> 
> ## **Your project is approved! 🎊 🥅 💛**
> **You may close this Pull Request and merge the changes**
> 
> **Keep being awesome devs and good luck with your next project!**
> 
> Cheers and Happy coding!👏👏👏
> 
> Code review by **Sarah Uryelah Silva Chamorro**
> 
> (Feel free to contact me on slack by the handle of **sarah** if you have any additional question)

Thank you, we've implemented the requested optional validations.

**Pull request merged**